modules/table/src/integrationTest/java/org/apache/ignite/internal/table/ItTxResourcesVacuumTest.java

```diff
@@ -0,0 +1,559 @@
+            + "}";
+
+    private ExecutorService txStateStorageExecutor = Executors.newSingleThreadExecutor();
+
+    @BeforeEach
+    @Override
+    public void setup(TestInfo testInfo) throws Exception {
+        super.setup(testInfo);
+
+        String zoneSql = "create zone test_zone with partitions=20, replicas=" + REPLICAS
+                + ", storage_profiles='" + DEFAULT_STORAGE_PROFILE + "'";
+        String sql = "create table " + TABLE_NAME + " (key bigint primary key, val varchar(20)) with primary_zone='TEST_ZONE'";
+
+        cluster.doInSession(0, session -> {
+            executeUpdate(zoneSql, session);
+            executeUpdate(sql, session);
+        });
+
+        txStateStorageExecutor = Executors.newSingleThreadExecutor(IgniteThreadFactory.create("test", "tx-state-storage-test-pool", log,
+                ThreadOperation.STORAGE_READ));
+    }
+
+    @Override
+    @AfterEach
+    public void tearDown() {
+        shutdownAndAwaitTermination(txStateStorageExecutor, 10, TimeUnit.SECONDS);
+
+        super.tearDown();
+    }
+
+    @Override
+    protected void customizeInitParameters(InitParametersBuilder builder) {
+        super.customizeInitParameters(builder);
+
+        builder.clusterConfiguration("{"
+                + "  transaction: {"
+                + "      implicitTransactionTimeout: 30000,"
+                + "      txnResourceTtl: 2"
+                + "  },"
+                + "  replication: {"
+                + "      rpcTimeout: 30000"
+                + "  },"
+                + "}");
+    }
+
+    @Override
+    protected int initialNodes() {
+        return 3;
+    }
+
+    /**
+     * Returns node bootstrap config template.
+     *
+     * @return Node bootstrap config template.
+     */
+    @Override
+    protected String getNodeBootstrapConfigTemplate() {
+        return NODE_BOOTSTRAP_CFG_TEMPLATE;
+    }
+
+    @Test
+    public void testVacuum() throws InterruptedException {
+        IgniteImpl node = anyNode();
+
+        RecordView<Tuple> view = node.tables().table(TABLE_NAME).recordView();
+
+        // Put some value into the table.
+        Transaction tx = node.transactions().begin();
+        UUID txId = txId(tx);
+
+        log.info("Test: Loading the data [tx={}].", txId);
+
+        Tuple tuple = findTupleToBeHostedOnNode(node, TABLE_NAME, tx, INITIAL_TUPLE, NEXT_TUPLE, true);
+
+        int partId = partitionIdForTuple(node, TABLE_NAME, tuple, tx);
+
+        Set<String> nodes = partitionAssignment(node, new TablePartitionId(tableId(node, TABLE_NAME), partId));
+
+        view.upsert(tx, tuple);
+
+        tx.commit();
+
+        log.info("Test: Tx committed [tx={}].", txId);
+
+        checkVolatileTxStateOnNodes(nodes, txId);
+        waitForTxStateReplication(nodes, txId, partId, 10_000);
+
+        waitForTxStateVacuum(txId, partId, true, 10_000);
+    }
+
+    @Test
+    public void testVacuumWithCleanupDelay() throws InterruptedException {
```

Comment:
Test name isn't fully self descriptive and test itself is a bit overcomplicated. Please share brief test explanation with javadoc, e.g. "Ensure that volatile txn state, removed by the vacuum, will be restored by txn cleanup finalization and fully removed by the next vacuum iteration." 
I'd also highly prefer several comments within test, something like following:
```
// Populate table with some data, hold txn cleanup for a while.
... 
some code goes here
...
// Trigger vacuum and ensure that volatile txn state was vacuumized despite the fact that txn cleanup wasn't finished yet.
... 
some code goes here
...
// Release txn cleanup and check that txn persistent state was updated with cleanupFinishedTimestamp.
... 
some code goes here
...
// Trigger one more vacuum iteration and ensure that both persistent and volatile states were vacuumized.
... 
some code goes here
...
// Read some data (with vacuumized txn states) with RW transaction.
...
// Read some data (with vacuumized txn states) with RO transaction with readTimestamp t25 (gt than tx1 commit timestamp)
...
```
And finally I'd put all that comments altogether within test steps in test javadoc, so basically it will look like following:
```
    /**
     * Ensure that volatile txn state, removed by the vacuum, will be restored by txn cleanup finalization and fully removed by the next vacuum iteration.
     *
     * <ul>
     *  <li>Populate table with some data, hold txn cleanup for a while.</li>
     *  <li>Trigger vacuum and ensure that volatile txn state was vacuumized despite the fact that txn cleanup wasn't finished yet.</li>
     * ...
     * </ul>
     */
@Test
void testVauumWithCleanupDelay() {
// Populate table with some data, hold txn cleanup for a while.
... 
some code goes here
...
// Trigger vacuum and ensure that volatile txn state was vacuumized despite the fact that txn cleanup wasn't finished yet.
... 
some code goes here
...
}
```

Reply:
Steps provided here are more about `testVacuumPersistentStateAfterCleanupDelayAndVolatileStateVacuum()` but it's not important.

Reply:
Added steps to javadocs.